File: src/network/packet/impl/PlayerSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;
use pocketcloud\cloud\player\CloudPlayer;

final class PlayerSyncPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(
        private readonly ?CloudPlayer $player = null,
        private readonly bool $removal = false
    ) {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->player, $this->removal);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getPlayer(): ?CloudPlayer {
        return $this->player;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public static function create(CloudPlayer $player, bool $removal): self {
        return new self($player, $removal);
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/PlayerSyncPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.player.CloudPlayer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class PlayerSyncPacket extends CloudPacket implements ClientboundPacket {

    private CloudPlayer player;
    private boolean removal;

    public PlayerSyncPacket(CloudPlayer player, boolean removal) {
        this.player = player;
        this.removal = removal;
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(player, removal);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static PlayerSyncPacket create(CloudPlayer player, boolean removal) {
        return new PlayerSyncPacket(player, removal);
    }
}
